### PR TITLE
arm/rp2040: Add SPI driver support

### PR DESCRIPTION
**Pull request by yunkya2**
## Summary
Add SPI driver for Raspberry Pi Pico.
(DMA transfer is not supported yet.)

## Impact
rp2040 only

## Testing
New config `raspberrypi-pico:spisd` is added to test SD card access with SPI.
Read/Write access OK.

### CODE REVIEW

File: arch/arm/src/rp2040/rp2040_gpio.h [hunk 1]
```diff
@@ -50,12 +66,44 @@ extern "C"
...
 
+/****************************************************************************
+ * Inline Functions
+ ****************************************************************************/
+
+static inline void rp2040_gpio_put(uint32_t gpio, int set)
+{
+  uint32_t value = 1 << gpio;
+
+  if (set)
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_gpio.h:77]: you missed
{
}
It is required, like in the MISRA C

File: arch/arm/src/rp2040/rp2040_gpio.h [hunk 2]
```diff
@@ -50,12 +66,44 @@ extern "C"
...
+
+static inline void rp2040_gpio_setdir(uint32_t gpio, int out)
+{
+  uint32_t value = 1 << gpio;
+
+  if (out)
+    putreg32(value, RP2040_SIO_GPIO_OE_SET);
+  else
+    putreg32(value, RP2040_SIO_GPIO_OE_CLR);
+}
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_gpio.h:98]: Ditto

File: arch/arm/src/rp2040/rp2040_spi.h [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+#ifdef CONFIG_RP2040_SPI1
+void  rp2040_spi1select(FAR struct spi_dev_s *dev,
+                       uint32_t devid,
+                       bool selected);
+uint8_t rp2040_spi1status(FAR struct spi_dev_s *dev,
+                         uint32_t devid);
+#ifdef CONFIG_SPI_CMDDATA
+int rp2040_spi1cmddata(FAR struct spi_dev_s *dev,
+                      uint32_t devid,
+                      bool cmd);
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_spi.h:163]: uint32_t and bool should start aligned below the FAR

File: arch/arm/src/rp2040/rp2040_spi.h [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+#ifdef CONFIG_RP2040_SPI0
+void  rp2040_spi0select(FAR struct spi_dev_s *dev,
+                       uint32_t devid,
+                       bool selected);
+uint8_t rp2040_spi0status(FAR struct spi_dev_s *dev,
+                         uint32_t devid);
+#ifdef CONFIG_SPI_CMDDATA
+int rp2040_spi0cmddata(FAR struct spi_dev_s *dev,
+                      uint32_t devid,
+                      bool cmd);
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_spi.h:150]: uint32_t and bool should start aligned below the FAR

File: arch/arm/src/rp2040/rp2040_spi.h [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+ *   rp2040_spi0/1cmddata must also be provided.
+ *   This functions performs cmd/data selection operations using GPIOs in
+ *   the way your board is configured.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RP2040_SPI0
+void  rp2040_spi0select(FAR struct spi_dev_s *dev,
+                       uint32_t devid,
+                       bool selected);
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_spi.h:144]: uint32_t and bool should start aligned below the FAR

File: boards/arm/rp2040/common/src/rp2040_spisd.c [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: board_spisd_status
+ *
+ * Description:
+ *   Get the status whether SD Card is present or not.
+ *   This function is called only from cxd56_spi.c.
```
Comment (masayuki2009) [inline boards/arm/rp2040/common/src/rp2040_spisd.c:115]: cxd56_spi.c -> rp2040_spi.c ?

**Review by masayuki2009 — CHANGES_REQUESTED** [review]

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**